RecoTauTag/RecoTau/plugins/RecoTauPiZeroStripPlugin3.cc

```diff
@@ -0,0 +1,393 @@
+  std::unique_ptr<TFormula> etaAssociationDistance_; // size of strip clustering window in eta direction
+  std::unique_ptr<TFormula> phiAssociationDistance_; // size of strip clustering window in phi direction
+
+  bool updateStripAfterEachDaughter_;
+  int maxStripBuildIterations_;
+
+  // Parameters for build strip combinations
+  bool combineStrips_;
+  int maxStrips_;
+  double combinatoricStripMassHypo_;
+
+  AddFourMomenta p4Builder_;
+
+  int verbosity_;
+};
+
+namespace
+{
+  std::unique_ptr<TFormula> makeFunction(const std::string& functionName, const edm::ParameterSet& pset)
+  {
+    TString formula = pset.getParameter<std::string>("function");
+    formula = formula.ReplaceAll("pT", "x");
+    std::unique_ptr<TFormula> function(new TFormula(functionName.data(), formula.Data()));
+    int numParameter = function->GetNpar();
+    for ( int idxParameter = 0; idxParameter < numParameter; ++idxParameter ) {
+      std::string parameterName = Form("par%i", idxParameter);
+      double parameter = pset.getParameter<double>(parameterName);
+      function->SetParameter(idxParameter, parameter);
+    }
+    return function;
+  }
+}
+
+RecoTauPiZeroStripPlugin3::RecoTauPiZeroStripPlugin3(const edm::ParameterSet& pset, edm::ConsumesCollector &&iC)
+  : RecoTauPiZeroBuilderPlugin(pset, std::move(iC)),
+    vertexAssociator_(pset.getParameter<edm::ParameterSet>("qualityCuts"), std::move(iC)), qcuts_(nullptr), etaAssociationDistance_(nullptr), phiAssociationDistance_(nullptr)
+{
+  minGammaEtStripSeed_ = pset.getParameter<double>("minGammaEtStripSeed");
+  minGammaEtStripAdd_ = pset.getParameter<double>("minGammaEtStripAdd");
+
+  minStripEt_ = pset.getParameter<double>("minStripEt");
+  
+  edm::ParameterSet qcuts_pset = pset.getParameterSet("qualityCuts").getParameterSet("signalQualityCuts");
+//-------------------------------------------------------------------------------
+// CV: disable track quality cuts for PFElectronsPFElectron
+//       (treat PFElectrons like PFGammas for the purpose of building eta-phi strips)
+  applyElecTrackQcuts_ = pset.getParameter<bool>("applyElecTrackQcuts");
+  if ( !applyElecTrackQcuts_ ) {
+    qcuts_pset.addParameter<double>("minTrackPt", std::min(minGammaEtStripSeed_, minGammaEtStripAdd_));
+    qcuts_pset.addParameter<double>("maxTrackChi2", 1.e+9);
+    qcuts_pset.addParameter<double>("maxTransverseImpactParameter", 1.e+9);
+    qcuts_pset.addParameter<double>("maxDeltaZ", 1.e+9);
+    qcuts_pset.addParameter<double>("minTrackVertexWeight", -1.);
+    qcuts_pset.addParameter<unsigned>("minTrackPixelHits", 0);
+    qcuts_pset.addParameter<unsigned>("minTrackHits", 0);
+  }
+//-------------------------------------------------------------------------------
+  qcuts_pset.addParameter<double>("minGammaEt", std::min(minGammaEtStripSeed_, minGammaEtStripAdd_));
+  //qcuts_ = new RecoTauQualityCuts(qcuts_pset);
+  std::unique_ptr<RecoTauQualityCuts> qcuts_(new RecoTauQualityCuts(qcuts_pset));
```

Comment:
this defines a local qcuts_ variable which dies when the function closes.
You should set the qcuts_ from  RecoTauPiZeroStripPlugin3 data members  ```qcuts_.reset(new RecoTauQualityCuts(qcuts_pset)); ```